cluster-autoscaler/cloudprovider/externalgrpc/externalgrpc_cloud_provider.go

```diff
@@ -324,30 +329,32 @@ func BuildExternalGrpc(
 	if err != nil {
 		klog.Fatalf("Could not open cloud provider configuration file %q: %v", opts.CloudConfig, err)
 	}
-	client, err := newExternalGrpcCloudProviderClient(config)
+	client, grpcTimeout, err := newExternalGrpcCloudProviderClient(config)
 	if err != nil {
 		klog.Fatalf("Could not create gRPC client: %v", err)
 	}
-	return newExternalGrpcCloudProvider(client, rl)
+	return newExternalGrpcCloudProvider(client, grpcTimeout, rl)
 }
 
 // cloudConfig is the struct hoding the configs to connect to the external cluster autoscaler provider service.
+// sigs.k8s.io/yaml actually reads the json tag
 type cloudConfig struct {
-	Address string `yaml:"address"` // external cluster autoscaler provider address of the form "host:port", "host%zone:port", "[host]:port" or "[host%zone]:port"
-	Key     string `yaml:"key"`     // path to file containing the tls key
-	Cert    string `yaml:"cert"`    // path to file containing the tls certificate
-	Cacert  string `yaml:"cacert"`  // path to file containing the CA certificate
+	Address     string           `json:"address"`                // external cluster autoscaler provider address of the form "host:port", "host%zone:port", "[host]:port" or "[host%zone]:port"
+	Key         string           `json:"key"`                    // path to file containing the tls key
+	Cert        string           `json:"cert"`                   // path to file containing the tls certificate
+	Cacert      string           `json:"cacert"`                 // path to file containing the CA certificate
+	GRPCTimeout *metav1.Duration `json:"grpc_timeout,omitempty"` // timeout of invoking a grpc call
```

Comment:
can you please also update the [readme](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler/cloudprovider/externalgrpc#configuration) with the new configuration key?